torchgeo/datasets/cropharvest.py

```diff
@@ -0,0 +1,323 @@
+            & (self.labels["properties.dataset"] == dataset)
+        ]
+        row = row.to_dict(orient="records")[0]
+        label = "None"
+        if row["properties.label"]:
+            label = row["properties.label"]
+        elif row["properties.is_crop"] == 1:
+            label = "Other"
+
+        return torch.tensor(np.where(self.classes == label)[0][0])
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        # Check if feature files already exist
+        feature_path = os.path.join(
+            self.root, self.file_dict["features"]["extracted_filename"]
+        )
+        feature_path_zip = os.path.join(
+            self.root, self.file_dict["features"]["filename"]
+        )
+        label_path = os.path.join(
+            self.root, self.file_dict["labels"]["extracted_filename"]
+        )
+        # Check if labels exist
+        if os.path.exists(label_path):
+            # Check if features exist
+            if os.path.exists(feature_path):
+                return
+            # Check if features are downloaded in zip format
+            if os.path.exists(feature_path_zip):
+                self._extract()
+                return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        # Download and extract the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset and extract it."""
+        features_path = os.path.join(self.file_dict["features"]["filename"])
+        download_url(
+            self.file_dict["features"]["url"],
+            self.root,
+            filename=features_path,
+            md5=self.file_dict["features"]["md5"] if self.checksum else None,
+        )
+
+        download_url(
+            self.file_dict["labels"]["url"],
+            self.root,
+            filename=os.path.join(self.file_dict["labels"]["filename"]),
+            md5=self.file_dict["labels"]["md5"] if self.checksum else None,
+        )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        features_path = os.path.join(self.root, self.file_dict["features"]["filename"])
+        extract_archive(features_path)
+
+    def plot(self, sample: dict[str, Tensor], subtitle: Optional[str] = None) -> Figure:
+        """Plot a sample from the dataset using bands for Agriculture RGB composite.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            suptitle: optional subtitle to use for figure
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        fig, axs = plt.subplots()
+        bands = [self.all_bands.index(band) for band in self.rgb_bands]
+        rgb = np.array(sample["array"])[:, bands]
+        normalized = percentile_normalization(rgb)
+        axs.imshow(normalized[None, ...])
```

Comment:
Still need to remove normalization here so the color matches during plotting

Reply:
All of my values for the B4, B3, and B2 bands are in the thousands. Is this a sign of a potential issue or could you clarify what you mean by removing the normalization?

Reply:
No, that's normal. Most satellites record values in 12 or 16 bits instead of 8 bits. In the Sentinel-2 dataset, we divide by 10K to get the values to a range that looks reasonable color-wise. Can you try that to see if the colors look more like we would expect (green for most crops)?

Reply:
<img width="622" alt="Screenshot 2024-01-16 at 6 09 29 PM" src="https://github.com/microsoft/torchgeo/assets/36140322/f644e7fe-f3f2-4889-b52d-cf38045dd0e3">
The plots are looking quite dark, could I add a fixed value to bring them up to a better range?

Reply:
Try dividing by 3K, that's what we did in SSL4EO.

Reply:
<img width="607" alt="Screenshot 2024-01-17 at 10 09 19 AM" src="https://github.com/microsoft/torchgeo/assets/36140322/0e626f72-7c02-4245-b1b4-372865aaf824">
<img width="626" alt="Screenshot 2024-01-17 at 10 18 13 AM" src="https://github.com/microsoft/torchgeo/assets/36140322/2708e6cb-5768-4f0b-bca6-5df48a037074">

is this looking closer to normal?

Reply:
Hmm, closer, but still weird. I would expect it to be pretty green for the whole middle of the year. Are we using the right bands? Does the raw data from the dataset include any normalization? Did you remove the normalization from the plot function and only include the division?

Reply:
A few interesting things:

- I double checked the bands being used and confirmed there is no normalization of the raw data or during plotting except for division, all seems to be good there
- It seems time-series in the dataset do not start at January, but rather a month defined in the metadata. I can update the plot accordingly but this will only shift the graph, though that might make a little more sense looking at the forage crop plot, knowing it starts in February
- I also believe the maize plot is an anomaly as the date the label was assigned occurred almost two years after the end of the time-series. After investigating further samples, this seems to be uncommon as most labels during or just around the time-series
<img width="380" alt="Screenshot 2024-01-19 at 12 08 04 AM" src="https://github.com/microsoft/torchgeo/assets/36140322/d0a3205e-5e8f-41c6-8f22-55302b1587e3">
<img width="616" alt="Screenshot 2024-01-19 at 12 21 52 AM" src="https://github.com/microsoft/torchgeo/assets/36140322/15b078be-046b-4fde-acd3-8ea3c2716b65">
<img width="634" alt="Screenshot 2024-01-19 at 12 21 59 AM" src="https://github.com/microsoft/torchgeo/assets/36140322/829ddf26-31fb-414f-8141-26b616cc555a">
<img width="636" alt="Screenshot 2024-01-19 at 12 22 15 AM" src="https://github.com/microsoft/torchgeo/assets/36140322/3a94014e-579d-48a7-95d5-b7fee6c82017">
<img width="624" alt="Screenshot 2024-01-19 at 12 22 22 AM" src="https://github.com/microsoft/torchgeo/assets/36140322/b8830300-9c4f-461c-af68-cd5a12b33714">

Reply:
Alright, those look less crazy. Let's go with this so as not to sink any more time into this dataset. Thanks for validating!